interview-questions.md

```diff
@@ -8,62 +8,62 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 1. What is iteration?
 
-  Your answer:
+  Your answer: An iteration is a movement through a function until a certain condition is met. Once the condition is met the iteration will end. 
 
-  Researched answer:
+  Researched answer: An iteration is the execution of code repetitively until a condition is met. Iterations are typically used for strings and arrays. 
 
 
 
 2. The JavaScript built in method .map() takes predefined parameters. What are they? Which are required and which are optional?
 
-  Your answer:
+  Your answer: The built-in method .map() takes in three predefined paramaters - value, index, and array, in that order. The value parameter is the only one required, however if you want to use the array parameter you will need to include the index parameter to get the array parameter. All parameters must be provided in the order of value, index, and array in order function properly. 
 
-  Researched answer:
+  Researched answer: The .map() function iterates through an array and must return an array of the same length, unlike filter which can return a smaller array. The built-in method .map() takes in three predefined paramaters - value, index, and array, in that order. The value parameter is the only one required, however if you want to use the array parameter you will need to include the index parameter to get the array parameter. All parameters must be provided in the order of value, index, and array in order function properly. 
 
 
 
 3. What is object destructuring?
 
-  Your answer:
+  Your answer: Object destructuring allows developers to take an object and make it into smaller, unique variables.  
 
-  Researched answer:
+  Researched answer: Object destructuring allows developers to take properties of an object and separate them into separate variables. Object destructuring creates a path to a particular key. Several keys can be assigned to the same destructuring shortcut. 
 
 
 
 4. What is the difference between a function and a method?
 
-  Your answer:
+  Your answer: Functions are reusable code that return an output. Methods are functions that belong to an object. 
 
-  Researched answer:
+  Researched answer:Functions are reusable code that return an output. Functions allow developers to reuse code, making it generic and dynamic so that different inputs can run through the function. Methods are functions that belong to an object. Therefore, methods use "this" within the object so the code understands that the method is calling on variables within the object. 
 
 
 
 5. What is the difference between a class and an object?
 
-  Your answer:
+  Your answer: Objects hold information like arrays. Classes are reusable collections of information. 
 
-  Researched answer:
+  Researched answer: Classes are a function that contain data and behavior. Objects hold collections of information, like arrays and other objects. An object has information and behavior and contain key value pairs.
```

Comment:
A class is a recipe/ blueprint/ plan for an object that can be utilized to create instances of of that class. These instances are the data type object